crates/polars-time/src/upsample.rs

```diff
@@ -119,6 +119,13 @@ fn upsample_impl(
     stable: bool,
 ) -> PolarsResult<DataFrame> {
     let s = source.column(index_column)?;
+    if offset.parsed_int || every.parsed_int {
+        polars_ensure!(
+            ((offset.parsed_int || offset.is_zero())
+             && (every.parsed_int || every.is_zero())),
+            ComputeError: "you cannot combine time durations like '2h' with integer durations like '3i'"
+        )
+    }
```

Comment:
@MarcoGorelli  I found the condition you suggested returns different behavior in the following cases:

```
Current:   (every="1h", offset="0i") --> Error
Suggested: (every="1h", offset="0i") --> OK
```

(I added related test cases to `test_upsample_index_invalid` to clarify the difference)

I borrowed the current condition from the group_by_dynamic's code, and changing only this condition causes a mismatch with the behavior of the group_by_dynamic's validation.
If we could fix this condition, I think it would be better to fix both at the same time.

Reply:
ah thanks for explaining - ok to keep in sync then